develop/toolkit/api-libraries/polkadart.md

```diff
@@ -0,0 +1,154 @@
+  final polkadot = Polkadot(provider);
+}
+```
+
+### 2. Reading Chain Data
+
+```dart
+final stateApi = StateApi(provider);
+
+// Get current runtime version
+final runtimeVersion = await stateApi.getRuntimeVersion();
+print(runtimeVersion.toJson());
+
+// Get metadata
+final metadata = await stateApi.getMetadata();
+print('Metadata version: ${metadata.version}');
+```
+
+### 3. Subscribe to New Blocks
+
+```dart
+final subscription = await provider.subscribe('chain_subscribeNewHeads', []);
+
+subscription.stream.forEach((response) {
+  print('New head: ${response.result}');
+});
+```
+
+### 4. Send a Transaction
+
+```dart
+final wallet = await KeyPair.sr25519.fromUri("//Alice");
+print('Alice\' wallet: ${wallet.address}');
+
+// Get information necessary to build a proper extrinsic
+final runtimeVersion = await polkadot.rpc.state.getRuntimeVersion();
+final currentBlockNumber = (await polkadot.query.system.number()) - 1;
+final currentBlockHash = await polkadot.query.system.blockHash(currentBlockNumber);
+final genesisHash = await polkadot.query.system.blockHash(0);
+final nonce = await polkadot.rpc.system.accountNextIndex(wallet.address);
+
+// Make the encoded call
+final multiAddress = $MultiAddress().id(wallet.publicKey.bytes);
+final transferCall = polkadot.tx.balances.transferKeepAlive(dest: multiAddress, value: BigInt.one).encode();
+
+// Make the payload
+final payload = SigningPayload(
+    method: transferCall,
+    specVersion: runtimeVersion.specVersion,
+    transactionVersion: runtimeVersion.transactionVersion,
+    genesisHash: encodeHex(genesisHash),
+    blockHash: encodeHex(currentBlockHash),
+    blockNumber: currentBlockNumber,
+    eraPeriod: 64,
+    nonce: nonce,
+    tip: 0,
+).encode(polkadot.registry);
+
+// Sign the payload and build the final extrinsic
+final signature = wallet.sign(payload);
+final extrinsic = ExtrinsicPayload(
+  signer: wallet.bytes(),
+  method: transferCall,
+  signature: signature,
+  eraPeriod: 64,
+  blockNumber: currentBlockNumber,
+  nonce: nonce,
+  tip: 0,
+).encode(polkadot.registry, SignatureType.sr25519);
+
+// Send the extrinsic to the blockchain
+final author = AuthorApi(provider);
+await author.submitAndWatchExtrinsic(extrinsic, (data) {
+  print(data);
+});
+```
+
+## Type Generation
+
+### 1. Run Generator
+
+```bash
+dart run polkadart_cli:generate -v
+```
+
+### 2. Use Generated Types
+
+```dart
+import 'package:your_app/generated/polkadot/polkadot.dart';
+import 'package:polkadart/polkadart.dart';
+import 'package:ss58/ss58.dart';
+
+final provider = Provider.fromUri(Uri.parse('wss://rpc.polkadot.io'));
+final polkadot = Polkadot(provider);
+  
+// Account from SS58 address
+final account = Address.decode('19t9Q2ay58hMDaeg6eeBhqmHsRnc2jDMV3cYYw9zbc59HLj');
+
+// Retrieve Account Balance
+final accountInfo = await polkadot.query.system.account(account.pubkey);
+print('Balance: ${accountInfo.data.free}')
+```
```

Comment:
Consider moving this up as the first example, since type generation is needed to interact with a Polkadot SDK chain.